snooty/postprocess.py

```diff
@@ -903,6 +1061,45 @@ def build_slug_fileid_mapping(self) -> None:
             fileid_dict[slug] = fileid
         self.slug_fileid_mapping = fileid_dict
 
+    def build_iatree(self) -> Dict[str, SerializableType]:
+        starting_page = self.pages.get(FileId("index.txt"))
+
+        if not starting_page:
+            return {}
+        if not isinstance(starting_page.ast, n.Root):
+            return {}
+        if not starting_page.ast.options.get("ia"):
+            return {}
+
+        title: Sequence[n.InlineNode] = self.heading_handler.get_title("index") or [
+            n.Text((0,), self.project_config.title)
+        ]
+        root: Dict[str, SerializableType] = {
+            "title": [node.serialize() for node in title],
+            "slug": "/",
+            "children": [],
+        }
+        self.iterate_ia(starting_page, root)
+        return root
+
+    def iterate_ia(self, page: Page, result: Dict[str, SerializableType]) -> None:
```

Comment:
Could we have a docstring to briefly explain the purpose and behavior of this method?